dependencies.yaml

```diff
@@ -56,6 +57,14 @@ dependencies:
           - scikit-build=0.17.1
           - tritonclient=2.26 # Required by NvTabular, force the version, so we get protobufs compatible with 4.21
           - ucx=1.14
+  development_cpp:
+    common:
+      - output_types: [conda]
+        packages:
+          - clangdev=16
+          - include-what-you-use=0.18
```

Comment:
IMPORTANT: Be sure to update `docker/conda/environments/cuda11.8_dev.yml` to match.

Also locally make change a cpp file and then run `ci/scripts/cpp_checks.sh` to verify that its working. I believe I tried this a while ago and ran into some trouble.